mlflow/keras.py

```diff
@@ -217,6 +217,9 @@ def _is_tf_keras(model):
     # save keras model to path/data/model.h5
     model_subpath = os.path.join(data_subpath, _MODEL_SAVE_PATH)
     model_path = os.path.join(path, model_subpath)
+    # if the path ends with ".h5", save will use hdf5
+    if "save_format" in kwargs and kwargs["save_format"] == "tf" and model_path.endswith(".tf"):
+        model_path = model_path[:-3]
```

Comment:
Right now this PR only seems to address how to save the model; what happens when we later try to load this model? If I understand correctly, the load path will still look for the file with `.h5` suffix; which the file would _not_ have if it were successfully saved with `save_format='tf'`

We probably also want to make sure that this change is backwards compatible (i.e. so that models with the `.h5` suffix can be loaded successfully with changes)